### PR TITLE
nnvp-42

### PR DESCRIPTION
**Pull request by mateusztheglue**

### CODE REVIEW

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-28** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-28.cluster.development.nnvp.thegluesandbox.com)

File: src/main/java/com/theglue/nnvp/entity/Vehicle.java [hunk 1]
```diff
@@ -29,12 +43,42 @@
 @Entity
 public class Vehicle extends BaseEntity {
 
+  @Size(max = 15)
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/entity/Vehicle.java]: in VehicleDTO it's 4

File: src/main/java/com/theglue/nnvp/enums/Traction.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public enum Traction implements NnvpEnum<Traction> {
+
+  TRACTION_1("1", "VOORWIELAANDRIJVING"),
+  TRACTION_2("2", "ACHTERWIELAANDRIJVING"),
+  TRACTION_3("3", "4 WIELAANDRIJVING");
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/enums/Traction.java]: general question 

maybe in such cases we could go for omitting the code and implementing (default?) method getCode with something like name.split(PREFIX_)[1] or so?

File: src/main/java/com/theglue/nnvp/enums/Injection.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public enum Injection implements NnvpEnum<Injection> {
+
+  D("D"),
+  Y("Y");
+
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/enums/Injection.java]: do we need this label field?

File: src/main/java/com/theglue/nnvp/enums/PlusMinus.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+@Getter
+public enum PlusMinus implements NnvpEnum<PlusMinus> {
+
+  PLUS("+", "+"),
+  MINUS("-", "-");
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/enums/PlusMinus.java]: again I guess we could live with one field here

File: src/main/java/com/theglue/nnvp/enums/ProctectionCode.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package com.theglue.nnvp.enums;
```
Comment (eryk7b) [inline src/main/java/com/theglue/nnvp/enums/ProctectionCode.java]: :D 
ProtectionCode vs ProctectionCode :P ???

**Comment by laurent-theglue** [comment]
:star: PR built and available in a preview environment **the-glue-nnvp-backend-pr-28** [here](http://nnvp-backend.jx-the-glue-nnvp-backend-pr-28.cluster.development.nnvp.thegluesandbox.com)

**Pull request merged**